### ISSUE TITLE
Optimize TFLite argmin/argmax

### ISSUE THREAD

**Issue by WindQAQ** [tensorflow/tensorflow#47555]
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Pixel 3a Android 11
- TensorFlow installed from (source or binary): source
- TensorFlow version (or github SHA if from source): nightly

**Standalone code to reproduce the issue** 

```python
model = tf.keras.Sequential([
  tf.keras.Input(shape=(512,), batch_size=1, dtype=tf.int32),
  tf.keras.layers.Embedding(1000, 128),
  tf.keras.layers.LSTM(64, return_sequences=True),
  tf.keras.layers.Dense(5000),
  tf.keras.layers.Lambda(lambda x: tf.math.argmax(x, axis=-1)),
])
```

Given a simple sequence model with 5000 output labels, ArgMax takes as much time as computation-intense ops like FullyConnected and LSTM. Here is my benchmark on pixel 3a

```
                     [Node type]          [count]         [avg ms]          [avg %]         [cdf %]       [mem KB]      [times called]
        UNIDIRECTIONAL_SEQUENCE_LSTM            1           27.552          33.447%         33.447%          0.000              1
                 FULLY_CONNECTED                1           27.474          33.353%         66.800%          0.000              1
                         ARG_MAX                1           24.868          30.189%         96.989%          0.000              1
                         RESHAPE                1            2.343           2.844%         99.834%          0.000              1
                          GATHER                1            0.137           0.166%        100.000%          0.000              1
```

Looks like compiler is not smart enough to vectorize naive-for-loop argmin-max reduction. In order to improve the performance, we should at least specialize the implementation when `axis=dims-1`, that is, `inner_size=1` in the following code snippet. If the request is valid, I can try to handcraft some Neon intrinsics to see if the performance is better.

https://github.com/tensorflow/tensorflow/blob/ea49c350b3b812bd852ae8456d2c8cc29aff6c95/tensorflow/lite/kernels/internal/reference/arg_min_max.h#L49-L63

**Comment by abattery** [comment]
Thanks for filing the feature request!

**Comment by WindQAQ** [comment]
@abattery Here is my benchmark on argmax of a 1-D float32 tensor because I only focus on `axis=-1`. For 2-D or higher tensor with `axis=-1`, it will roughly take `outer_size * timeof(axis_size)` time.

On Pixel 3a android 11.

| Size | Naive (min/max/avg microsecs) | Hand-crafted Neon intrinsics (min/max/avg microsecs) |
| ---- | -------| ------------ |
| 10 | 0/63/0.260737| 0/49/0.216432  |
| 100 | 1/101/1.17919   | 0/77/0.271518 |
| 1000 | 9/98/9.80665 | 0/65/1.07653  |
| 10000 | 95/190/96.2557  | 8/110/9.63559 |
| 100000 | 952/1102/967.445  | 88/215/94.8766 |

Let me know if it's acceptable!

**Comment by abattery** [comment]
Looks like it can achieve 10x improvement. Very impressive! Look forward to seeing the PR soon.
